clients/da-vinci-client/src/main/java/com/linkedin/davinci/store/AbstractStorageEngine.java

```diff
@@ -234,6 +227,16 @@ public synchronized void addStoragePartition(StoragePartitionConfig storageParti
 
     Partition partition = createStoragePartition(storagePartitionConfig);
     this.partitionList.set(partitionId, partition);
+    if (this.rwLockForStoragePartitionAdjustmentList.get(partitionId) == null) {
```

Comment:
nit: could be putIfAbsent or computeIfAbsent for better atomicity - although current implementation is not wrong..

Reply:
This is a list, not a map, so it doesn't have these functions. I've been thinking about building a proper / full featured array-based map for use cases like these, but I didn't get around to it yet...

Reply:
Oh right, I forget this is not a map😂